src/reducers/logProcessor/resmoke.js

```diff
@@ -28,7 +28,7 @@ export default function(state: Log, response: string): Log {
   const gitPrefixLen = gitPrefix.length + 2;
   let gitVersionStr: string = 'master';
   const portRegex = / [sdbc](\d{1,5})\|/;
-  const stateRegex = /(:shard\d*|:configsvr)?:(initsync|primary|mongos|secondary\d*|node\d*)]/;
+  const stateRegex = /(:s(hard)?\d*|:c(onfigsvr)?)?:(initsync|prim(ary)?|(mongo)?s|sec(ondary)?\d*|n(ode)?\d*)]/;
```

Comment:
Would it make sense to add a non-alphanumeric match pattern at the end of each capture group? For example if you take just the `(mongo)?s|sec(ondary)?` pattern, the word "steven" is a match simply because it has an "s" in it

Reply:
We have hardly noticeable `]` after the `(initsync|prim(ary)?|(mongo)?s|sec(ondary)?\d*|n(ode)?\d*)` group for that purpose, so it won't find something like `steven`. And also there is `:` before that group.

I've double checked, it will find only if it would be something like `:s]teven`.